File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/country-region-currency/country-region-currency-lookup-model.ts

export interface ICountryRegionCurrencyLookupModel {
  countryRegionCode: string;
  currencyCode: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class CountryRegionCurrencyLookupModel implements ICountryRegionCurrencyLookupModel {
  countryRegionCode: string;
  currencyCode: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<CountryRegionCurrencyLookupModel>) {
    if (!!init) {
      Object.assign<CountryRegionCurrencyLookupModel, Partial<CountryRegionCurrencyLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Sales/CountryRegionCurrency/Commands/Queries/GetCountryRegionCurrencies/CountryRegionCurrencyLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Person.CountryRegion.Queries.GetCountryRegions;

namespace AdventureWorks.Application.DataEngine.Sales.CountryRegionCurrency.Queries.GetCountryRegionCurrencies
{
    public partial class CountryRegionCurrencyLookupModel : IHaveCustomMapping
    {
        public string CountryRegionCode { get; set; }
        public string CurrencyCode { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.CountryRegionCurrency, CountryRegionCurrencyLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<CountryRegionCurrencyLookupModel, BaseCountryRegionCurrency>().IgnoreMissingDestinationMembers();
        }
    }
}
